### PR TITLE
perlfunc: add general description of goto

### LINKED ISSUE #21003: [doc] Documentation for 'goto &NAME' should mention that it does not return
**Where**
perlfunc -f goto

**Description**
'goto &NAME' does not return. The documentation should mention this explicitly.

### PR DESCRIPTION
**Pull request by mauke**
Fixes #21003.

### CODE REVIEW

File: pod/perlfunc.pod [hunk 1]
```diff
@@ -3582,6 +3582,9 @@ X<goto> X<jump> X<jmp>
 
 =for Pod::Functions create spaghetti code
 
+C<goto> transfers execution to a specified point in the program. Unlike a
+subroutine call, it does not return.
```
Comment (demerphq) [inline pod/perlfunc.pod]: Am i being excessively pedantic when I think that `goto &sub;` does return, just not to where it was goto'ed from? I think this should be 'goto LABEL' only.

Comment (jkeenan) [reply pod/perlfunc.pod]: @mauke, can you respond to @demerphq's comment above?  Thanks.

Comment (mauke) [reply pod/perlfunc.pod]: `goto` is a control transfer. Execution flows into `goto`, but does not come out again. That's why I think it is correct to say "goto does not return".
(If you use `goto &sub`, the subroutine does return eventually, of course, but `goto` itself does not.)

Comment (mauke) [reply pod/perlfunc.pod]: I've reworded it, hopefully making it less ambiguous.

**Review by khwilliamson — APPROVED** [review]

**Pull request merged**